apps/hash-api/src/integrations/google/shared/get-or-check-access-token.ts

```diff
@@ -0,0 +1,89 @@
+import {
+  createGoogleOAuth2Client,
+  getTokensForGoogleAccount,
+} from "@local/hash-backend-utils/google";
+import { Request, Response } from "express";
+
+import { enabledIntegrations } from "../../enabled-integrations";
+import { getGoogleAccountById } from "./get-google-account";
+
+/**
+ * Shared function to retrieve a Google access token for an Express request,
+ * providing standard error handling and response handling.
+ * Calling controllers decide what to do with the token.
+ */
+export const getGoogleAccessTokenForExpressRequest = async ({
+  googleAccountId,
+  req,
+  res,
+}: {
+  googleAccountId: string;
+  req: Request;
+  res: Response;
+}) => {
+  if (!req.user) {
+    res.status(401).send({ error: "User not authenticated." });
+    return;
+  }
+
+  if (!req.context.vaultClient) {
+    res.status(501).send({ error: "Vault integration is not configured." });
+    return;
+  }
+
+  if (!enabledIntegrations.googleSheets) {
+    res.status(501).send({ error: "Google integration is not enabled." });
+    return;
+  }
```

Comment:
I'm not sure it's worth it, because we'd have to do one of:
1. Pass in the `res` to the function so that it can send the appropriate response if something is missing, which makes it harder to follow the response handling since part of it is buried in another function (I do this in `get-or-check-access-token` and I'm not a big fan of it there either, but that's saving a lot more repeated code)
2. When we get the `user` and `vaultClient` back, in this function send a response if one is missing, in which case we'd have the same amount of repeated code here

Reply:
Alright fair enough, happy to keep as is then